pkg/migration/migrations/raft_applied_index_term.go

```diff
@@ -51,8 +75,30 @@ func raftAppliedIndexTermMigration(
 			if bytes.Compare(desc.StartKey, keys.LocalMax) < 0 {
 				start, _ = keys.Addr(keys.LocalMax)
 			}
-			if err := deps.DB.Migrate(ctx, start, end, cv.Version); err != nil {
+
+			// Skip any ranges below the resume watermark.
```

Comment:
This is more to remind myself: by comparison against end key, is it possible to skip any ranges? Comparing against the end key is safe because the effects of the below-raft migration are retained for a keyspan across range splits and merges, right? If a migrated range splits off, we're relying on the RHS to also use post-migration state (in fact, new ranges should typically start off at that state -- these migrations are only to clear out stragglers). If a migrated range is merged into (it is the LHS pre-merge, and subsumes the right-most neighbour), because it's a below-raft migration, the RHS keyspan is "automatically merged". We're also transacting over a consistent view of meta{1,2} so there's no risk of a range split/merge happening from underneath us. Ok, LGTM.

Reply:
Correct. If a `Migrate` command succeeds across some key span, then all ranges within that span are migrated, regardless of any splits or merges. We compare with the end key rather than the start key to make sure we handle e.g. splits.

> If a migrated range is merged into (it is the LHS pre-merge, and subsumes the right-most neighbour), because it's a below-raft migration, the RHS keyspan is "automatically migrated".

That may be true, but it doesn't matter: if the LHS was migrated but the RHS wasn't, then the new end key of the range is the RHS' end key, so when we retry the migration we'll apply the migrate command on the merged range.

Reply:
That was was more to walk through what would happen if the range was merged and then later split, not just whether the retry happened post-merge. But yea this all feels safe.